### PR TITLE
Fix TPC price issue and debug info

### LINKED ISSUE #2889: TPC Amount Field Values for New Modifiers Jump to Input in Next Row
Steps to reproduce:

- start free ticket and add default prices
- click to add new modifier after base price
- click in amount field for new price modifier
- type 1 on keyboard
- amount of default text in next row changes to 1
- new modifier amount field shows nothing
- tax field that changed was also disabled because it's a default tax

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR:
- Fixes the price input issue cause due to index being used as key.
- Improves memoization in `/ticketPriceCalculator/components/table/Table.tsx` to avoid object literals going into props
- Fixes the issue with `DebugInfo` component
- Closes #2889

### CODE REVIEW

File: assets/src/application/ui/display/DebugInfo.tsx [hunk 1]
```diff
@@ -31,18 +32,16 @@ const DebugInfo: React.FC<DebugInfoProps> = ({ data, asJson = true, asCollapse =
 
 	const output = <Pre>{dataToRender}</Pre>;
 
+	// define it here to avoid conditional call of hook
+	const handleToggle = useCallback(() => setShow((v) => !v), [setShow]);
 	if (!asCollapse) {
 		return output;
 	}
 	const buttonText = show ? __('Hide Debug Info') : __('Show Debug Info');
-	const handleToggle = useCallback(() => setShow(!show), [show]);
```
Comment (manzoorwanijk) [inline assets/src/application/ui/display/DebugInfo.tsx:38]: This has happened again 😄

File: assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/components/table/Table.tsx [hunk 2]
```diff
@@ -23,20 +24,34 @@ const Table: React.FC<TableProps> = ({ prices }) => {
...
 	return (
 		<ResponsiveTable
 			bodyRows={bodyRows}
-			className={{ tableClassName: 'ee-ticket-price-calculator' }}
-			footerRows={[footerRow]}
-			headerRows={[headerRow]}
-			metaData={{
-				tableId: 'ticket-price-calculator-table',
-				tableCaption: __('Ticket Price Calculator'),
-			}}
```
Comment (manzoorwanijk) [inline assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/components/table/Table.tsx:39]: Passing object literals as props are a big reason for rerenders

File: assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/components/table/useBodyRowGenerator.tsx [hunk 3]
```diff
@@ -61,7 +61,7 @@ const useBodyRowGenerator = (): BodyRowGenerator => {
 			cells,
 			className: `ee-editor-date-list-view-row`,
 			id: `ee-editor-date-list-view-row-${price.id}`,
-			key: `row-${index}`,
+			key: `row-${price.id}`,
```
Comment (manzoorwanijk) [inline assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/components/table/useBodyRowGenerator.tsx:64]: Here is the bug that caused the issue.

File: assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/components/table/Table.tsx (hunk 2 above)
Comment (tn3rb) [reply assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/components/table/Table.tsx:39]: ahhhh... ya that makes sense... that's my bad

File: assets/src/application/ui/display/DebugInfo.tsx (hunk 1 above)
Comment (nicolad) [reply assets/src/application/ui/display/DebugInfo.tsx:38]: isn't there an eslint rule for that?

File: assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/components/table/useBodyRowGenerator.tsx (hunk 3 above)
Comment (tn3rb) [reply assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/components/table/useBodyRowGenerator.tsx:64]: good find

File: assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/components/table/Table.tsx [hunk 4]
```diff
@@ -12,6 +12,7 @@ import { useDataState } from '../../data';
 import { useMoneyDisplay } from '@appServices/utilities/money';
 
 import './styles.scss';
+import { useMemoStringify } from '@application/services/hooks';
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/components/table/Table.tsx:15]: :sunglasses:

File: assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/components/table/Table.tsx [hunk 5]
```diff
@@ -23,20 +24,34 @@ const Table: React.FC<TableProps> = ({ prices }) => {
...
+	const headerRows = useMemo(() => {
+		const headerRow = headerRowGenerator({ signB4 });
+		return [headerRow];
+	}, [headerRowGenerator, signB4]);
+
+	const className = useMemoStringify({ tableClassName: 'ee-ticket-price-calculator' });
+	const metaData = useMemoStringify({
+		tableId: 'ticket-price-calculator-table',
+		tableCaption: __('Ticket Price Calculator'),
+	});
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/components/table/Table.tsx:46]: noice

File: assets/src/application/ui/display/DebugInfo.tsx (hunk 1 above)
Comment (manzoorwanijk) [reply assets/src/application/ui/display/DebugInfo.tsx:38]: > what's happened again?

Conditional calling of React Hook

>isn't there an eslint rule for that?

Yeah, react-hooks-eslint works, but not for us 🤷

File: assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/components/table/Table.tsx (hunk 2 above)
Comment (nicolad) [reply assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/components/table/Table.tsx:39]: I like to extract this to a variable and reference here after that.

**Pull request merged**